serveradmin/api/decorators.py

```diff
@@ -77,19 +84,22 @@ def _wrapper(request):
             if isinstance(error, ObjectDoesNotExist):
                 status_code = 404
 
+            logger.error('{} request data: {}'.format(str(error), {
+                'now': now,
+                'body': body,
+                'public_keys': public_keys,
+                'signatures': signatures,
+                'app_id': app_id,
+                'token': token,
+                'then': then,
+                'body_json': body_json,
+            }), exc_info=True)
             return_value = {
                 'error': {
                     'message': str(error),
                 }
             }
 
-        logger.info('api: Call: ' + (', '.join([
-            'Method: {}'.format(view.__name__),
-            'Application: {}'.format(app),
-            'Time elapsed: {:.3f}s'.format(
-                (timezone.now() - now).total_seconds()
-            ),
-        ])))
```

Comment:
This can stay where it is if you just log the error above.

Reply:
Correct me if I am wrong but I don't thinks so because it would potentially cause a NameError in python because authenticate_app might fail without declaring and setting the app variable which would then fail here.

Reply:
True, sorry.